rustls/Cargo.toml

```diff
@@ -129,6 +128,7 @@ rustdoc-args = ["--cfg", "docsrs"]
 [package.metadata.cargo_check_external_types]
 allowed_external_types = [
   # ---
+  "portable_atomic_util::arc::Arc",
```

Comment:
extra-minor nit: I think this should be done in a separate CI testing update, like what I did in other PR #52, where most CI jobs are repeated with both `alloc::sync::Arc` and `portable_atomic_util::Arc`.